northstar/src/runtime/island/seccomp.rs

```diff
@@ -13,18 +13,31 @@
 //   limitations under the License.
 
 use crate::{
-    npk::manifest::{Capability, Profile},
+    common::non_null_string::NonNullString,
+    npk::manifest::{Capability, Profile, SyscallArgRule, SyscallRule},
     runtime::island::seccomp_profiles::default,
 };
 use bindings::{
-    seccomp_data, sock_filter, sock_fprog, BPF_ABS, BPF_JEQ, BPF_JMP, BPF_K, BPF_LD, BPF_MAXINSNS,
-    BPF_RET, BPF_W, SECCOMP_RET_ALLOW, SECCOMP_RET_KILL, SECCOMP_RET_LOG, SYSCALL_MAP,
+    seccomp_data, sock_filter, sock_fprog, BPF_ABS, BPF_ALU, BPF_AND, BPF_IMM, BPF_JEQ, BPF_JMP,
+    BPF_K, BPF_LD, BPF_MAXINSNS, BPF_MEM, BPF_NEG, BPF_OR, BPF_RET, BPF_ST, BPF_W,
+    SECCOMP_RET_ALLOW, SECCOMP_RET_KILL, SECCOMP_RET_LOG, SYSCALL_MAP,
 };
-use log::warn;
+use log::trace;
 use nix::errno::Errno;
-use std::collections::HashSet;
+use std::{
+    collections::{HashMap, HashSet},
+    mem::size_of,
+};
 use thiserror::Error;
 
+// Required for platform check
+#[cfg(any(
+    not(any(target_arch = "aarch64", target_arch = "x86_64")),
+    target_pointer_width = "32",
+    target_endian = "big"
+))]
+use crate::runtime::island::seccomp::Error::UnsupportedPlatform;
```

Comment:
Can we make this a compile time error?

https://doc.rust-lang.org/std/macro.compile_error.html